modules/websearch/lib/websearch_webcoll.py

```diff
@@ -1166,29 +1174,43 @@ def task_run_core():
                 colls.append(get_collection(row[0]))
         # secondly, update collection reclist cache:
         if task_get_option('part', 1) == 1:
-            i = 0
-            for coll in colls:
-                i += 1
+            for i, coll in enumerate(colls):
                 write_message("%s / reclist cache update" % coll.name)
                 if str(coll.dbquery).startswith("hostedcollection:"):
                     coll.set_nbrecs_for_external_collection()
                 else:
                     coll.calculate_reclist()
                 coll.update_reclist()
-                task_update_progress("Part 1/2: done %d/%d" % (i, len(colls)))
+                task_update_progress("Part 1/2: done %d/%d" % (i + 1, len(colls)))
                 task_sleep_now_if_required(can_stop_too=True)
         # thirdly, update collection webpage cache:
         if task_get_option("part", 2) == 2:
-            i = 0
-            for coll in colls:
-                i += 1
+            ## Let's refresh the cache of restricted records, in order to
+            ## not disclose them in latest_addition.
+            from search_engine import restricted_collection_cache, get_all_restricted_recids
+            restricted_collection_cache.recreate_cache_if_needed()
+            all_restricted_recids = get_all_restricted_recids()
+
+            for i, coll in enumerate(colls):
                 if coll.reclist_updated_since_start or task_has_option("collection") or task_get_option("force") or not task_get_option("quick"):
                     write_message("%s / webpage cache update" % coll.name)
                     for lang in CFG_SITE_LANGS:
-                        coll.update_webpage_cache(lang)
+                        if coll.get_name() in restricted_collection_cache.cache:
+                            ## We don't strip away restricted records as the collection
+                            ## is anyway restricted and hence the user must be
+                            ## already authenticated to see records.
+                            ## Additionally, by default, since CFG_WEBSEARCH_VIEWRESTRCOLL_POLICY
+                            ## is set to ANY, any record belonging to a restricted
+                            ## collection is entitled to be displayed in the
+                            ## latest additions.
```

Comment:
@kaplun it might be that the fix/handling this correctly is tightly related to issue #1511 which actually would be quite important to solve.